cmd/lncli/commands.go

```diff
@@ -155,12 +161,22 @@ func newAddress(ctx *cli.Context) error {
 	// Map the string encoded address type, to the concrete typed address
 	// type enum. An unrecognized address type will result in an error.
 	stringAddrType := ctx.Args().First()
+	unused:= ctx.String("unused")
+
 	var addrType lnrpc.AddressType
 	switch stringAddrType { // TODO(roasbeef): make them ints on the cli?
 	case "p2wkh":
-		addrType = lnrpc.AddressType_WITNESS_PUBKEY_HASH
+		if unused == "true" {
```

Comment:
I'd just set the default type first and overwrite it with the "unused" one if set. That way you don't need an `else` case, which makes it a bit more readable IMO.